lib/lambda_ethereum_consensus/validator/setup.ex

```diff
@@ -0,0 +1,82 @@
+defmodule LambdaEthereumConsensus.Validator.Setup do
+  @moduledoc """
+  Module that setups the initial validators state
+  """
+
+  require Logger
+  alias LambdaEthereumConsensus.Validator
+
+  @spec init(Types.slot(), Types.root()) :: %{Bls.pubkey() => Validator.state()}
+  def init(slot, head_root) do
+    config = Application.get_env(:lambda_ethereum_consensus, __MODULE__, [])
+    keystore_dir = Keyword.get(config, :keystore_dir)
+    keystore_pass_dir = Keyword.get(config, :keystore_pass_dir)
+
+    setup_validators(slot, head_root, keystore_dir, keystore_pass_dir)
+  end
+
+  defp setup_validators(_s, _r, keystore_dir, keystore_pass_dir)
+       when is_nil(keystore_dir) or is_nil(keystore_pass_dir) do
+    Logger.warning(
+      "[Validator] No keystore_dir or keystore_pass_dir provided. Validator will not start."
+    )
+
+    %{}
+  end
+
+  defp setup_validators(slot, head_root, keystore_dir, keystore_pass_dir) do
+    validator_keys = decode_validator_keys(keystore_dir, keystore_pass_dir)
+
+    validators =
+      validator_keys
+      |> Enum.map(fn {pubkey, privkey} ->
+        {pubkey, Validator.new({slot, head_root, {pubkey, privkey}})}
+      end)
+      |> Map.new()
+
+    Logger.info("[Validator] Initialized #{Enum.count(validators)} validators")
+
+    validators
+  end
+
+  @doc """
+    Get validator keys from the keystore directory.
+    This function expects two files for each validator:
+      - <keystore_dir>/<public_key>.json
+      - <keystore_pass_dir>/<public_key>.txt
+  """
+  @spec decode_validator_keys(binary(), binary()) ::
+          list({Bls.pubkey(), Bls.privkey()})
+  def decode_validator_keys(keystore_dir, keystore_pass_dir)
+      when is_binary(keystore_dir) and is_binary(keystore_pass_dir) do
+    File.ls!(keystore_dir)
+    |> Enum.map(fn filename ->
```

Comment:
Yes totally agree, this hasn't changed really, it's marked as change because of the rename and github doesn't track that so good on the diffs. In git though, it shows this was the same as the previous iteration. 

That said I totally agree with you and I even already refactored this in the PR I'm working on ^^, I had this 2 already extracted:

```elixir
  defp decode_key({keystore_file, keystore_pass_file}) do
    # TODO: remove `try` and handle errors properly
    try do
      Keystore.decode_from_files!(keystore_file, keystore_pass_file)
    rescue
      error ->
        Logger.error(
          "[Validator] Failed to decode keystore file: #{keystore_file}. Pass file: #{keystore_pass_file} Error: #{inspect(error)}"
        )

        nil
    end
  end

  defp paths_from_filename(keystore_dir, keystore_pass_dir, filename, ".json") do
    basename = Path.basename(filename, ".json")

    keystore_file = Path.join(keystore_dir, "#{basename}.json")
    keystore_pass_file = Path.join(keystore_pass_dir, "#{basename}.txt")

    {keystore_file, keystore_pass_file}
  end

  defp paths_from_filename(_keystore_dir, _keystore_pass_dir, basename, _ext) do
    Logger.warning("[Validator] Skipping file: #{basename}. Not a json keystore file.")
    nil
  end
```